### PR TITLE
Swap on cowswap

### PR DESCRIPTION
**Pull request by gabrielfior**

### CODE REVIEW

**Comment by gabrielfior** [comment]
I'm waiting on the update of admin rights on the repo in order to add an ENV for the tests to pass.

File: .github/workflows/python_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Python CI
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+env:
+  MANIFOLD_API_KEY: ${{ secrets.MANIFOLD_API_KEY }}
```
Comment (kongzii) [inline .github/workflows/python_ci.yaml]: Most of these can be removed, or can't they?

File: cowswap_python_client/cow_client.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        )
+        return quote
+
+    @staticmethod
+    def _if_error_log_and_raise(r: requests.Response) -> None:
+        try:
+            r.raise_for_status()
+        except:
+            logger.error(f"Error occured on response: {r.text}")
+            r.raise_for_status()
```
Comment (kongzii) [inline cowswap_python_client/cow_client.py]:
```suggestion
            raise
```

No need to commitl, just fyi that you can just do `raise` to re-raise an exception in except block.

File: mypy.ini [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+show_traceback = True
+disallow_incomplete_defs = True
+disallow_untyped_defs = True
+ignore_missing_imports = True
+
+# See https://github.com/python/mypy/issues/3905#issuecomment-421065323
+# We don't want to ignore all missing imports as we want to catch those in our own code
+# But for certain libraries they don't have a stub file, so we only enforce import checking for our own libraries.
+# Another alternative would be to list out every single dependency that does not have a stub.
+[mypy-prediction_market_agent.*]
```
Comment (kongzii) [inline mypy.ini]:
```suggestion
[mypy-cowswap_python_client.*]
```

File: mypy.ini [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ignore_missing_imports = False
+[mypy-scripts.*]
```
Comment (kongzii) [inline mypy.ini]:
```suggestion
```

File: mypy.ini [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ignore_missing_imports = False
```
Comment (kongzii) [inline mypy.ini:23]:
```suggestion
```

**Review by kongzii — COMMENTED** [review]
Instead of `cowswap_python_client` I'd go with just `cowswap`, just to make it more user friendly, because doing

```
import cowswap_python_client
```

feels odd, of course it's python. 

But I just noticed that there is https://pypi.org/project/cowswap/ already! :o 

So maybe just cowswap_client? (unless you want to get in touch with that guy)

File: .github/workflows/python_ci.yaml (hunk 1 above)
Comment (gabrielfior) [reply .github/workflows/python_ci.yaml]: removed

File: cowswap_python_client/cow_client.py (hunk 2 above)
Comment (gabrielfior) [reply cowswap_python_client/cow_client.py]: Thanks, added this - it's nice because the exception traceback is kept.

**Review by kongzii — APPROVED** [review]

**Pull request merged**